lib/spack/spack/ci.py

```diff
@@ -2020,25 +2020,51 @@ def process_command(name, commands, repro_dir, run=True, exit_on_failure=True):
 
     Returns: the exit code from processing the command
     """
-    tty.debug("spack {0} arguments: {1}".format(name, commands))
-
+    tty.debug(f"spack {name} arguments: {commands}")
     if len(commands) == 0 or isinstance(commands[0], str):
         commands = [commands]
 
-    # Create a string [command 1] && [command 2] && ... && [command n] with commands
-    # quoted using double quotes.
-    args_to_string = lambda args: " ".join('"{}"'.format(arg) for arg in args)
-    full_command = " \n ".join(map(args_to_string, commands))
+    except_hook = """
+def global_except_hook(exctype, value, traceback):
+    if exctype == BaseException:
+        # Suppress all exceptions
+        pass
+    else:
+        sys.__excepthook__(exctype, value, traceback)
```

Comment:
A few things:

* What doesn't fall under BaseException that you want to let through?
* I think if this strategy persists,
  * you should store a reference to the original `sys.excepthook` and use that vs. calling `sys.__excepthook__`
  * Add a comment about what this is doing e.g. `# suppress all exceptions`
* I would think that generally using `os.system` to run commands shouldn't throw exceptions unless the commands are malformed, which is the sort of thing we'd want to correct, so overall I'm not clear on what cases this is supposed to handle

Reply:
Looping context from Slack thread:
This was an artifact from a previous iteration, it's not longer necessary and is being removed.